sw/device/lib/dif/dif_clkmgr.c

```diff
@@ -42,6 +42,19 @@ static bool clkmgr_measure_ctrl_regwen(const dif_clkmgr_t *clkmgr) {
                              CLKMGR_MEASURE_CTRL_REGWEN_EN_BIT);
 }
 
+dif_result_t dif_clkmgr_external_clock_is_settled(const dif_clkmgr_t *clkmgr,
```

Comment:
btw did this actually work? This is completely untested at the moment. In hindsight, i also wondered if i should have just made this a boolean instead of the full multi-bit..

Reply:
Good point about making it a straight boolean: this is not stored, and if anyone wants to fake the result they will either think the transition is done and will get recoverable errors, or will think the transition is not done and delay any action waiting for it.

By the way, thanks for asking: it made me realize I had not used this dif in the code. I changed the code to use it, and things work just fine.

Reply:
I added a spin wait till this function returns a positive status, and it works.